core/src/main/java/io/crate/types/ByteType.java

```diff
@@ -71,7 +71,21 @@ public Byte value(Object value) {
         if (val < Byte.MIN_VALUE || Byte.MAX_VALUE < val) {
             throw new IllegalArgumentException("byte value out of range: " + val);
         }
-        return (byte) val;
+        Byte byteValue = ((Number) value).byteValue();
```

Comment:
Can you fix the boxing warnings here and in the other places?

Reply:
I don't get any warnings here. Boxing is unavoidable because we have to return the boxed variant but the casting only works unboxed. Pushed a fix to reduce the boxing to once unboxing, once boxing.